Source/Scene/Model.js

```diff
@@ -2972,11 +3069,58 @@ define([
                     castShadows : castShadows,
                     receiveShadows : receiveShadows,
                     uniformMap : uniformMap,
-                    renderState : rs,
+                    renderState : drawRS,
                     owner : owner,
                     pass : isTranslucent ? Pass.TRANSLUCENT : Pass.OPAQUE
                 });
 
+                // Setup the stencil command for the highlight
+                var highlightRS = clone(rs);
+                highlightRS.stencilTest = {
+                    enabled : true,
+                    frontFunction : WebGLConstants.NOTEQUAL,
+                    backFunction : WebGLConstants.NOTEQUAL,
+                    reference : 1,
+                    mask : ~0,
+                    frontOperation : {
+                        fail : WebGLConstants.KEEP,
+                        zFail : WebGLConstants.KEEP,
+                        zPass : WebGLConstants.REPLACE
+                    },
+                    backOperation : {
+                        fail : WebGLConstants.KEEP,
+                        zFail : WebGLConstants.KEEP,
+                        zPass : WebGLConstants.REPLACE
+                    }
+                };
+                highlightRS.cull = {
+                    enabled : false,
+                    face : WebGLConstants.FRONT
+                };
+                highlightRS.depthTest = false;
+
+                highlightRS = RenderState.fromCache(highlightRS);
+
+                // Setup the highlight color and size uniforms.
+                uniformMap.u_highlightColor = createHighlightColorFunction(model);
+                uniformMap.u_highlightSize = createHighlightSizeFunction(model);
+
+                var highlightCommand = new DrawCommand({
```

Comment:
@lilleyse let me know if you have any questions on this one; I prefer creating these on-demand since they may not be used in many apps, and users should only pay for what they use when it's reasonable for us to implement.